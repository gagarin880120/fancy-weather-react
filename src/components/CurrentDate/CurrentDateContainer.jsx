import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import CurrentDate from './CurrentDate';
import { selectCurrentDate } from '../../redux/selectors';

export function CurrentDateContainer({ currentDate }) {
  return (
    <CurrentDate
      currentDate={currentDate}
    />
  );
}

export const mapStateToProps = (state) => ({
  currentDate: selectCurrentDate(state),
});

CurrentDateContainer.propTypes = {
  currentDate: PropTypes.string,
};

CurrentDateContainer.defaultProps = {
  currentDate: '',
};

export default connect(mapStateToProps)(CurrentDateContainer);
