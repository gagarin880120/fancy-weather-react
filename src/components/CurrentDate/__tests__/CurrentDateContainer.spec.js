import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { CurrentDateContainer, mapStateToProps } from '../CurrentDateContainer';

describe('CurrentDateContainer', () => {
  describe('component', () => {
    test('should render without crashing', () => {
      const renderer = new ShallowRenderer();
      renderer.render(<CurrentDateContainer />);

      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    test('should return right value', () => {
      const initialState = {
        currentDate: '2020-07-20 15:09:14',
        language: 'en',
      }

      expect(mapStateToProps(initialState)).toEqual({currentDate: 'Mon, 20 July, 15:09'});
    });
  });
});
