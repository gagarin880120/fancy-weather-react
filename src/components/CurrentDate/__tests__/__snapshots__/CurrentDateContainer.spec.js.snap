// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentDateContainer component should render without crashing 1`] = `
<CurrentDate
  currentDate=""
/>
`;
